knowledge-base/editor-add-icons.md

```diff
@@ -0,0 +1,197 @@
+---
+title: Add Icons in Editor
+description: How to insert icon tags in Editor?
+type: how-to
+page_title: Add Icons in Editor
+slug: editor-kb-add-icons
+position: 
+tags: editor, blazor, icon, schema, prosemirror
+ticketid: 
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Editor for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+I want to add custom icons in the Editor content. When adding an icon such as `<i class="fa-light fa-envelope">`, the icon's HTML disappears upon saving.
+
+How can I add an `<i>` element inside the Editor?
+
+## Solution
+
+The built-in ProseMirror schema that the Editor uses does not support `<i>` tag. To allow adding icons in the Editor content:
+
+1. [Modify the ProseMirror schema]({%slug editor-modify-default-schema%}) to include a node for the `<i>` element.
+2. Ensure the CSS classes for the icons are applied. The approach varies depending on the [edit mode of the Editor]({%slug editor-edit-modes-overview%})
+    * [Iframe Edit Mode](#add-icons-in-iframe-edit-mode)
+    * [Div Edit Mode](#add-icons-in-div-edit-mode)
+
+
+### Add icons in Iframe Edit Mode
+
+When the [EditMode of the Editor is set to `EditorEditMode.Iframe`]({%slug editor-edit-modes-iframe%}), the content area is an editable `<iframe>` element that does not inherit the CSS rules from the current page.
+
+This means that you need to inject the icons stylesheet into the `<iframe>`, so the icons are properly rendered. [This will be supported in a future UI for Blazor version](https://feedback.telerik.com/blazor/1543925-add-the-ability-to-inject-css-files-into-the-iframe). For the time-being you can inject the stylesheet with JSInterop in the `OnAfterRenderAsync`.
```

Comment:
The IFrame is the default edit mode and I believe a lot of people use it. That said, I'd prefer to give the customers an option they can use now as it may be a while before we implement the feature. We even used this approach in the docs - https://docs.telerik.com/blazor-ui/components/editor/prosemirror-plugins#adding-a-custom-plugin.
I realize this part will be outdated when we release the feature but this is why I've linked the feature request there. I may add a not indicating the version the kb was written in.